lib/puppet/parser/functions/defined_with_params.rb

```diff
@@ -24,7 +24,19 @@
     params = {}
   end
   ret = false
-  if resource = findresource(reference.to_s)
+
+  # Workaround for PE-20308
+  if reference.is_a?(String)
+    type_name, title = Puppet::Resource.type_and_title(reference, nil)
+  elsif reference.is_a?(Puppet::Resource)
+    type_name, title = Puppet::Resource.type_and_title(reference.type, reference.title)
```

Comment:
I had been looking for something like that originally, but overlooked it. Thanks!